src/com/oltpbenchmark/DistributionStatistics.java

```diff
@@ -1,179 +1,170 @@
-	public int getCount() {
-		return count;
-	}
-
-	public double getAverage() {
-		return average;
-	}
-
-	public double getStandardDeviation() {
-		return standardDeviation;
-	}
-
-	public double getMinimum() {
-		return percentiles[MINIMUM];
-	}
-
-	public double get25thPercentile() {
-		return percentiles[PERCENTILE_25TH];
-	}
-
-	public double getMedian() {
-		return percentiles[MEDIAN];
-	}
-
-	public double get75thPercentile() {
-		return percentiles[PERCENTILE_75TH];
-	}
-
-	public double get90thPercentile() {
-		return percentiles[PERCENTILE_90TH];
-	}
-
-	public double get95thPercentile() {
-		return percentiles[PERCENTILE_95TH];
-	}
-
-	public double get99thPercentile() {
-		return percentiles[PERCENTILE_99TH];
-	}
-
-	public double getMaximum() {
-		return percentiles[MAXIMUM];
-	}
-
-	@Override
-	public String toString() {
-		// convert times to ms
-		return "[min=" + getMinimum() / 1e6 + ", " + "25th="
-				+ get25thPercentile() / 1e6 + ", " + "median="
-				+ getMedian() / 1e6 + ", " + "avg=" + getAverage() / 1e6 + ", "
-				+ "75th=" + get75thPercentile() / 1e6 + ", " + "90th="
-				+ get90thPercentile() / 1e6 + ", " + "95th="
-				+ get95thPercentile() / 1e6 + ", " + "99th="
-				+ get99thPercentile() / 1e6 + ", " + "max=" + getMaximum()
-				/ 1e6 + "]";
-	}
-
-	public Map<String, Double> toMap() {
-		Map<String, Double> distMap = new LinkedHashMap<String, Double>();
-		distMap.put("Minimum Latency (milliseconds)", getMinimum() / 1e3);
-		distMap.put("25th Percentile Latency (milliseconds)", get25thPercentile() / 1e3);
-		distMap.put("Median Latency (milliseconds)", getMedian() / 1e3);
-		distMap.put("Average Latency (milliseconds)", getAverage() / 1e3);
-		distMap.put("75th Percentile Latency (milliseconds)", get75thPercentile() / 1e3);
-		distMap.put("90th Percentile Latency (milliseconds)", get90thPercentile() / 1e3);
-		distMap.put("95th Percentile Latency (milliseconds)", get95thPercentile() / 1e3);
-		distMap.put("99th Percentile Latency (milliseconds)", get99thPercentile() / 1e3);
-		distMap.put("Maximum Latency (milliseconds)", getMaximum() / 1e3);
-		return distMap;
-	}
+    private static final Logger LOG = LoggerFactory.getLogger(DistributionStatistics.class);
+
+    private static final double[] PERCENTILES = {0.0, 0.25, 0.5, 0.75, 0.9, 0.95, 0.99, 1.0};
+
+    private static final int MINIMUM = 0;
+    private static final int PERCENTILE_25TH = 1;
+    private static final int MEDIAN = 2;
+    private static final int PERCENTILE_75TH = 3;
+    private static final int PERCENTILE_90TH = 4;
+    private static final int PERCENTILE_95TH = 5;
+    private static final int PERCENTILE_99TH = 6;
+    private static final int MAXIMUM = 7;
+
+    private final int count;
+    private final long[] percentiles;
+    private final double average;
+    private final double standardDeviation;
+
+    public DistributionStatistics(int count, long[] percentiles, double average, double standardDeviation) {
+        this.count = count;
```

Comment:
TODO again with the asserts.. probably last time I'll comment on it. If we want them we can grep them out and add them back in manually.